Content.Client/Atmos/UI/GasAnalyzerWindow.xaml.cs

```diff
@@ -0,0 +1,115 @@
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.Utility;
+using Content.Shared.Temperature;
+using static Content.Shared.Atmos.Components.SharedGasAnalyzerComponent;
+
+namespace Content.Client.Atmos.UI
+{
+    /// <summary>
+    /// Gas analyzer UI.
+    /// </summary>
+    [GenerateTypedNameReferences]
+    public partial class GasAnalyzerWindow : SS14Window
+    {
+        int _lastGasCount = 0;
+
+        public GasAnalyzerWindow()
+        {
+            RobustXamlLoader.Load(this);
+        }
+        public void UpdateState(GasAnalyzerBoundUserInterfaceState state)
+        {
+            GasGrid.RemoveAllChildren();
+
+            var statusMessage = new FormattedMessage();
+            StatusLabel.SetMessage(statusMessage);
+
+            if (state.Error != null)
+            {
+                GasBar.RemoveAllChildren();
+                statusMessage.PushColor(Color.Red);
+                statusMessage.AddMarkup(Loc.GetString("gas-analyzer-window-error-text", ("errorText", state.Error)));
+                statusMessage.Pop();
+                return;
+            }
+
+            statusMessage.AddMarkup(Loc.GetString("gas-analyzer-window-pressure-text", ("pressure", $"{state.Pressure:0.##}")));
+            statusMessage.PushNewline();
+            statusMessage.AddMarkup(Loc.GetString("gas-analyzer-window-temperature-text",
+                                     ("tempK", $"{state.Temperature:0.#}"),
+                                     ("tempC", $"{TemperatureHelpers.KelvinToCelsius(state.Temperature):0.#}")));
+
+            if(state.Gases == null)
+            {
+                GasBar.RemoveAllChildren();
+                return;
+            }
+
+            var totalGasAmount = 0f;
+            foreach (var gas in state.Gases)
+            {
+                totalGasAmount += gas.Amount;
+            }
+
+            var minSize = 24; // This basically allows gases which are too small, to be shown properly
```

Comment:
I recognize that we do this in the current gas analyzer, but I still don't think it's the right thing to do. I'd much rather have accurate sizes, and some sort of callout at the end for quantities too small to be shown.

Reply:
Something like this (forgive the quick inkscape mockup):

![callout](https://user-images.githubusercontent.com/602406/147485969-ec0c1f62-33a9-4aa9-a502-6b67adff5cce.png)

Reply:
Thoughts, @Zumorica?

Reply:
That would be neat, yeah